src/scripts/sidebar.js

```diff
@@ -86,16 +105,66 @@ class SideBar extends EventEmitter {
         item.appendChild(hotkey);
 
         item.dataset.host = host.url;
+        item.dataset.sortOrder = hostOrder;
         item.setAttribute('server', host.url);
         item.classList.add('instance');
 
+        item.setAttribute('draggable', true);
+
+        item.ondragstart = (event) => {
+            window.dragged = event.target.nodeName !== 'LI' ? event.target.closest('li') : event.target;
+            event.dataTransfer.effectAllowed = 'move';
+            event.dataTransfer.dropEffect = 'move';
+            event.target.style.opacity = .5;
+        };
+
+        item.ondragover = (event) => {
+            event.preventDefault();
+        };
+
+        item.ondragenter = (event) => {
+            if (this.isBefore(window.dragged, event.target)) {
+                event.currentTarget.parentNode.insertBefore(window.dragged, event.currentTarget);
+            } else if (event.currentTarget !== event.currentTarget.parentNode.lastChild) {
+                event.currentTarget.parentNode.insertBefore(window.dragged, event.currentTarget.nextSibling);
+            } else {
+                event.currentTarget.parentNode.appendChild(window.dragged);
+            }
+        };
+
+        item.ondragend = (event) => {
+            event.target.style.opacity = '';
+        };
+
+        item.ondrop = (event) => {
+            event.preventDefault();
+
+            const newSortOrder = [];
+            Array.from(event.currentTarget.parentNode.children)
+                .map((sideBarElement) => {
+                    const url = sideBarElement.dataset.host;
+                    newSortOrder.push(url);
+                    this.remove(url);
+
+                    return sideBarElement;
+                })
+                .map((sideBarElement) => {
+                    this.sortOrder = newSortOrder;
+                    localStorage.setItem(this.sortOrderKey, JSON.stringify(this.sortOrder));
+
+                    const url = sideBarElement.dataset.host;
+                    const host = { url, title: sideBarElement.querySelector('div.tooltip').innerHTML };
```

Comment:
We should add the `active` class to the selected server as well.

Reply:
done! it should be after recreating the servers on the sidebar, so I put it on the end of the drop listener